### PR TITLE
`azurerm_lighthouse_definition` - support the `eligible_authorization` property

### PR DESCRIPTION
**Pull request by neil-yechenwei**
This PR is to support new property `eligible_authorization` for `azurerm_lighthouse_definition`.

Note: As the TCs related with Lighthouse Definition are skipped in Teamcity due to test data issue, so I ran them on my local and they passed.

--- PASS: TestAccLighthouseDefinition_plan (134.77s)
--- PASS: TestAccLighthouseDefinition_basic (135.99s)
--- PASS: TestAccLighthouseDefinition_emptyID (138.56s)
--- PASS: TestAccLighthouseDefinition_eligibleAuthorization (157.81s)
--- PASS: TestAccLighthouseDefinition_complete (160.86s)
--- PASS: TestAccLighthouseDefinition_requiresImport (184.04s)
--- PASS: TestAccLighthouseDefinition_update (527.25s)

### CODE REVIEW

File: internal/services/lighthouse/lighthouse_definition_resource.go [hunk 1]
```diff
@@ -99,6 +100,76 @@ func resourceLighthouseDefinition() *pluginsdk.Resource {
...
+										Optional: true,
+										Elem: &pluginsdk.Resource{
+											Schema: map[string]*pluginsdk.Schema{
+												"principal_id": {
+													Type:         pluginsdk.TypeString,
+													Required:     true,
+													ValidateFunc: validation.IsUUID,
+												},
+
+												"principal_display_name": {
```
Comment (jackofallops) [inline internal/services/lighthouse/lighthouse_definition_resource.go:152]: Does this provide any additional benefit to just using the `principal_id`? What happens if the id and display name do not match?

File: internal/services/lighthouse/lighthouse_definition_resource.go [hunk 2]
```diff
@@ -99,6 +100,76 @@ func resourceLighthouseDefinition() *pluginsdk.Resource {
 				Optional: true,
 			},
 
+			"eligible_authorization": {
+				Type:     pluginsdk.TypeSet,
+				Optional: true,
+				Elem: &pluginsdk.Resource{
```
Comment (jackofallops) [inline internal/services/lighthouse/lighthouse_definition_resource.go:106]: Are there a maximum number of elements this Set allows in the API?

File: website/docs/r/lighthouse_definition.html.markdown [hunk 3]
```diff
@@ -64,6 +66,38 @@ An `authorization` block supports the following:
...
+
+~> **Note:** When this property isn't set, it would be set to `None`.
+
+* `maximum_activation_duration` - (Optional) The maximum access duration in ISO 8601 format for just-in-time access requests. Defaults to `PT8H`.
+
+* `approver` - (Optional) An `approver` block as defined below.
+
+---
+
+A `approver` block supports the following:
```
Comment (jackofallops) [inline website/docs/r/lighthouse_definition.html.markdown:93]:
```suggestion
An `approver` block supports the following:
```

File: internal/services/lighthouse/lighthouse_definition_resource.go [hunk 4]
```diff
@@ -99,6 +100,76 @@ func resourceLighthouseDefinition() *pluginsdk.Resource {
...
+													ValidateFunc: validation.StringIsNotEmpty,
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+
+						"principal_display_name": {
```
Comment (jackofallops) [inline internal/services/lighthouse/lighthouse_definition_resource.go:164]: as above, what happens if this does not match the display name of the `principal_id`?

File: internal/services/lighthouse/lighthouse_definition_resource_test.go [hunk 5]
```diff
@@ -383,3 +404,50 @@ resource "azurerm_lighthouse_definition" "test" {
...
+  managing_tenant_id       = "%s"
+  scope                    = data.azurerm_subscription.test.id
+
+  authorization {
+    principal_id           = "%s"
+    role_definition_id     = data.azurerm_role_definition.reader.role_definition_id
+    principal_display_name = "Reader"
+  }
+
+  eligible_authorization {
```
Comment (jackofallops) [inline internal/services/lighthouse/lighthouse_definition_resource_test.go:436]: Should we have a test with multiple entries for this new block?

**Review by jackofallops — COMMENTED** [review]
Thanks @neil-yechenwei 
This looks off to a good start, some questions below on the new schema if you could take a look.

Thanks!

File: internal/services/lighthouse/lighthouse_definition_resource.go (hunk 2 above)
Comment (neil-yechenwei) [reply internal/services/lighthouse/lighthouse_definition_resource.go:106]: I didn't find the maximum number from anywhere. The [`autorization`](https://github.com/hashicorp/terraform-provider-azurerm/blob/8f106f76d3af98a79f29d53c74cfd19adaa0e183/internal/services/lighthouse/lighthouse_definition_resource.go#L61) previously added also didn't mention the maximum number.

File: internal/services/lighthouse/lighthouse_definition_resource.go (hunk 4 above)
Comment (neil-yechenwei) [reply internal/services/lighthouse/lighthouse_definition_resource.go:164]: If the id and display name do not match, the resource still can be created successfully. Per my understanding, it's used to better identify the principal. To align with [the previous style](https://github.com/hashicorp/terraform-provider-azurerm/blob/8f106f76d3af98a79f29d53c74cfd19adaa0e183/internal/services/lighthouse/lighthouse_definition_resource.go#L79), so I added it here.

File: internal/services/lighthouse/lighthouse_definition_resource_test.go (hunk 5 above)
Comment (neil-yechenwei) [reply internal/services/lighthouse/lighthouse_definition_resource_test.go:436]: As test data limitation, there is no more principal for testing. [As you can see](https://github.com/hashicorp/terraform-provider-azurerm/pull/19499#pullrequestreview-1201643682), we only can test them on local for test data limitation issue.

File: website/docs/r/lighthouse_definition.html.markdown (hunk 3 above)
Comment (neil-yechenwei) [reply website/docs/r/lighthouse_definition.html.markdown:93]: Updated

**Comment by neil-yechenwei** [comment]
@jackofallops , thanks for the comment. I've updated PR and left some suggestion.

**Review by katbyte — APPROVED** [review]
LGTM ⛵

**Pull request merged**